### PR TITLE
Typescript clases Resolves #279

### LINKED ISSUE #279: Add TypeScript Classes
# Description

Classes for TypeScript aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Requirements

* Copy `web/thesauruses/_meta/classes.json` to the right TypeScript directory in `web/thesauruses/`
* Edit the classes.json file to accurately reflect the latest major version of TypeScript
* Delete the `"name":` fields before your PR
* Add in `"comment":` if you wish to explain something in the code
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then leave `"code":` blank.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by alexindris**
## What GitHub issue does this PR apply to?

Resolves #279

## What changed and why?

Typescript classes,json

## Checklist

- [x] I claimed any associated issue(s) and they are not someone else's
- [x] I have looked at documentation to ensure I made any revisions correctly
- [x] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

**Comment by alexindris** [comment]
I have also found some errors in the clases meta can i open a issue and if you want i can fix it?

**Comment by geekygirlsarah** [comment]
> 
> 
> I have also found some errors in the clases meta can i open a issue and if you want i can fix it?

Yeah! The meta language files usually involve more discussion (as I try to look at a higher level across many languages) but I'd love to have any errors or problems brought up so I/we/someone can fix them!

File: web/thesauruses/typescript/classes.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    },
+    "extending_interface": {
+      "code": "interface IPerson {\n name: string;\n gender: string; \n}\n interface IEmployee extends IPerson {\n empCode: number; \n}"
+    },
+    "calling_superclass_functions": {
+      "code": "class Person{\n name:string\n eat():void{\n console.log(`${this.name} eats when hungry.`)\n }\n }\n class Student extends Person{\n rollnumber:number;\n constructor(rollnumber:number, name1:string){\n super(); // calling Parent's constructor\n }\n } "
+    },
+    "overriding_superclass_functions": {
+      "code": "class Person{\n name:string\n eat():void{\n console.log(`${this.name} eats when hungry.`)\n }\n }\n class Student extends Person{\n rollnumber:number;\n constructor(rollnumber:number, name1:string){\n super(); \n this.rollnumber = rollnumber\n this.name = name1\n }\n // overriding eat function\n eat():void{\n console.log(`${this.name} eats during break.`)\n }\n } "
+    },
```
Comment (geekygirlsarah) [inline web/thesauruses/typescript/classes.json:103]: Heh, interesting examples

**Comment by geekygirlsarah** [comment]
Alright, I think this looks good. I can merge it in.

Be sure to sign up for Hacktoberfest if you haven't for this to count towards the shirt/stickers!

**Pull request merged**